staging/src/k8s.io/client-go/dynamic/dynamicinformer/informer_test.go

```diff
@@ -95,7 +95,10 @@ func TestFilteredDynamicSharedInformerFactory(t *testing.T) {
 			if ts.existingObj != nil {
 				objs = append(objs, ts.existingObj)
 			}
-			fakeClient := fake.NewSimpleDynamicClient(scheme, objs...)
+			gvrToListKind := map[schema.GroupVersionResource]string{
+				{Group: "apps", Version: "v1", Resource: "deployments"}: "DeploymentList",
+			}
+			fakeClient := fake.NewSimpleDynamicClientWithCustomResourceMapping(scheme, gvrToListKind, objs...)
```

Comment:
sorry, i'm a bit confused. 'deployments' -> 'DeploymentList' should be guessable, no? is the switch from NewSimpleDynamicClient to NewSimpleDynamicClientWithCustomResourceMapping here really necessary?

Reply:
> sorry, i'm a bit confused. 'deployments' -> 'DeploymentList' should be guessable, no? is the switch from NewSimpleDynamicClient to NewSimpleDynamicClientWithCustomResourceMapping here really necessary?

The scheme they choose to test with does not have the deployment listed.  This test is specifically about whether dynamic resources (those without scheme entries) can be properly handled.  This means that it is correct for the usage.

Reply:
clarifying in comments.